docs/module.rst

```diff
@@ -0,0 +1,267 @@
+of which use templates. The template files can be made available in the local ``templates`` directory::
+
+    Project L
+        |─myModules.nf
+        └─templates
+            |─P1-template.sh
+            └─P2-template.sh
+
+Then, we have a second project A with a workflow that includes P1 and P2::
+
+    Pipeline A
+        └-main.nf
+
+Finally, we have a third project B with a workflow that also includes P1 and P2::
+
+    Pipeline B
+        └-main.nf
+
+With the possibility to keep the template files inside the project L, A and B can use the modules defined in L without any changes.
+A future project C would do the same, just cloning L (if not available on the system) and including its module script.
+
+Beside promoting the sharing of modules across pipelines, there are several advantages to keeping the module template under the script path:
+
+1. module components are *self-contained*,
+2. module components can be tested independently from the pipeline(s) that import them,
+3. it is possible to create libraries of module components.
+
+Ultimately, having multiple template locations allows a more structured organization within the same project. If a project
+has several module components, and all of them use templates, the project could group module scripts and their templates as needed. For example::
+
+    baseDir
+        |─main.nf
+        └─Phase0-Modules
+            |─mymodules1.nf
+            |─mymodules2.nf
+            └─templates
+                |─P1-template.sh
+                |─P2-template.sh
+        └─Phase1-Modules
+            |─mymodules3.nf
+            |─mymodules4.nf
+            └─templates
+                |─P3-template.sh
+                └─P4-template.sh
+        └─Phase2-Modules
+            |─mymodules5.nf
+            |─mymodules6.nf
+            └─templates
+                |─P5-template.sh
+                |─P6-template.sh
+                └─P7-template.sh
+
+Module binaries
+---------------
+
+.. note::
+    This feature requires Nextflow version ``22.10.0`` or later.
+
+Modules can define binary scripts that are locally scoped to the processes defined by the tasks.
+
+To enable this feature, enable the following flag in your pipeline script or configuration file::
+
+    nextflow.enable.moduleBinaries = true
+
+The binary scripts must be placed in the module directory names ``<module-dir>/resources/usr/bin``::
+
+    <module-dir>
+        |─main.nf
+        └─resources
+            └─usr
+                └─bin
+                    |─your-module-script1.sh
+                    └─another-module-script2.py
+
+Those scripts will be made accessible like any other command in the task environment, provided they have been granted
+the Linux execute permissions.
+
+.. note::
+    This feature requires the use of a local or shared file system for the pipeline work directory, or
+    :ref:`wave-page` when using container-native executors.
```

Comment:
Would be good to explicitly say what _won't_ work here. Not entirely sure if we're saying that it doesn't work with Docker? Or bucket storage? Or both?

Also should be a warning and not a note.

```suggestion
.. warning::
    This feature requires the use of a local or shared file system for the pipeline work directory.
    It will not work with container technologies such as Docker and work directories stored in buckets.
    To use module binaries in these situations you can use :ref:`wave-page`.
```

Reply:
This one is my bad, originally it said "cloud-based executors" and I changed it to "container-native" because I thought it was more accurate, but now I think I was wrong. I will change it back to "cloud-based".